inference-engine/include/cpp/ie_infer_request.hpp

```diff
@@ -64,41 +33,34 @@ class CompletionCallbackWrapper<IInferRequest::CompletionCallback> : public ICom
  * It can throw exceptions safely for the application, where it is properly handled.
  */
 class InferRequest {
-    IInferRequest::Ptr actual;
-    InferenceEngine::details::SharedObjectLoader::Ptr plg;
-    std::shared_ptr<details::ICompletionCallbackWrapper> callback;
+    std::shared_ptr<IInferRequestInternal>          _impl;
```

Comment:
do we need to keep `IInferRequestInternal` in public interface now?

Reply:
Just forward declaration.